hoomd/md/NeighborList.cc

```diff
@@ -1742,6 +1764,253 @@ void NeighborList::updateMemoryMapping()
+            for (unsigned int i = 0; i < m_exec_conf->getNRanks(); i++)
+                {
+                global_n_elem->push_back(0);
+                }
+            recvbuf = global_n_elem->data();
+            }
+
+        // Use a gather command to get the number of pairs on each processor
+        MPI_Gather(&n_elem, 1, MPI_INT, recvbuf, 1, MPI_INT, 0, m_exec_conf->getMPICommunicator());
+
+        // Allocate array to hold the offsets
+        std::vector<int>* offsets;
+        int* recvcount = nullptr;
+        int* displs = nullptr;
+        int total_elements = 0;
+        if (root)
+            {
+            offsets = new std::vector<int>();
+            offsets->reserve(m_exec_conf->getNRanks());
+            offsets->push_back(0);
+            total_elements = global_n_elem->data()[0];
+            for (unsigned int i = 1; i < m_exec_conf->getNRanks(); i++)
+                {
+                offsets->push_back(offsets->data()[i - 1] + global_n_elem->data()[i - 1]);
+                total_elements += global_n_elem->data()[i];
+                }
+            recvcount = global_n_elem->data();
+            displs = offsets->data();
+            }
+
+        // Use a gatherv command to produce the global_pair_list
+        if (root)
+            {
+            global_pair_list = (uint32_t*)malloc(sizeof(uint32_t) * total_elements);
+            free_when_done = pybind11::capsule(global_pair_list,
+                                               [](void* f)
+                                               {
+                                                   auto data = reinterpret_cast<uint32_t*>(f);
+                                                   free(data);
+                                               });
+            }
+        shape[0] = total_elements / 2;
+
+        MPI_Gatherv(pair_list->data(),
+                    n_elem,
+                    MPI_UINT32_T,
+                    global_pair_list,
+                    recvcount,
+                    displs,
+                    MPI_UINT32_T,
+                    0,
+                    m_exec_conf->getMPICommunicator());
+
+        // cleanup allocations
+        delete pair_list;
+        if (root)
+            {
+            delete global_n_elem;
+            delete offsets;
+            }
+        }
+    else
+        {
+        global_pair_list = (uint32_t*)pair_list->data();
+        shape[0] = pair_list->size();
+        free_when_done
+            = pybind11::capsule(pair_list,
+                                [](void* f)
+                                {
+                                    auto data = reinterpret_cast<std::vector<vec2<uint32_t>>*>(f);
+                                    delete data;
+                                });
+        }
+
+    if (root)
+        {
+        return pybind11::array_t(shape,            // shape
+                                 stride,           // stride
+                                 global_pair_list, // data pointer
```

Comment:
```suggestion
        return pybind11::array_t(shape,
                                 stride,
                                 global_pair_list,
```
The comments are redundant with the names of the parameters.